core/src/main/java/org/jdbi/v3/core/mapper/ColumnMappers.java

```diff
@@ -172,19 +172,21 @@ private ColumnMappers register(QualifiedColumnMapperFactory factory) {
      * @return a ColumnMapper for the given type, or empty if no column mapper is registered for the given type.
      */
     @Beta
-    public Optional<ColumnMapper<?>> findFor(QualifiedType type) {
+    public <T> Optional<ColumnMapper<T>> findFor(QualifiedType<T> type) {
         // ConcurrentHashMap can enter an infinite loop on nested computeIfAbsent calls.
         // Since column mappers can decorate other column mappers, we have to populate the cache the old fashioned way.
         // See https://bugs.openjdk.java.net/browse/JDK-8062841, https://bugs.openjdk.java.net/browse/JDK-8142175
-        ColumnMapper<?> cached = cache.get(type);
+        @SuppressWarnings("unchecked")
```

Comment:
Should this `@SuppressWarnings` be on the whole method?  There's a second cast slightly further down.

Reply:
I'm very much against giving suppressions a wide scope, they may end up hiding actual problems or things you _could_ improve if only you noticed them.